Formula/libclang.rb

```diff
@@ -0,0 +1,139 @@
+      system "make", "libclang"
+      system "make", "install"
+    end
+
+    (share/"clang/tools").install Dir["tools/clang/tools/scan-{build,view}"]
+  end
+
+  test do
+    assert_equal prefix.to_s, shell_output("#{bin}/llvm-config --prefix").chomp
+
+    (testpath/"sample.cpp").write <<~EOS
+      int main("aa", /* "bb"= */1 )
+    EOS
+
+    (testpath/"libclangtest.cpp").write <<~EOS
+      #include <stdio.h>
+      #include <stdlib.h>
+      #include <clang-c/Index.h>
+
+      const char *_getTokenKindSpelling(CXTokenKind kind) {
+        switch (kind) {
+          case CXToken_Punctuation: return "Punctuation"; break;
+          case CXToken_Keyword:     return "Keyword"; break;
+          case CXToken_Identifier:  return "Identifier"; break;
+          case CXToken_Literal:     return "Literal"; break;
+          case CXToken_Comment:     return "Comment"; break;
+          default:                  return "Unknown"; break;
+        }
+      }
+
+      unsigned get_filesize(const char *fileName) {
+        FILE *fp = fopen(fileName, "r");
+        fseek(fp, 0, SEEK_END);
+        auto size = ftell(fp);
+        fclose(fp);
+        return size;
+      }
+
+      CXSourceRange get_filerange(const CXTranslationUnit &tu, const char *filename) {
+        CXFile file = clang_getFile(tu, filename);
+        auto fileSize = get_filesize(filename);
+
+        CXSourceLocation topLoc  = clang_getLocationForOffset(tu, file, 0);
+        CXSourceLocation lastLoc = clang_getLocationForOffset(tu, file, fileSize);
+        if (clang_equalLocations(topLoc,  clang_getNullLocation()) ||
+            clang_equalLocations(lastLoc, clang_getNullLocation()) ) {
+          printf("cannot retrieve location\\n");
+          exit(1);
+        }
+
+        CXSourceRange range = clang_getRange(topLoc, lastLoc);
+        if (clang_Range_isNull(range)) {
+          printf("cannot retrieve range\\n");
+          exit(1);
+        }
+
+        return range;
+      }
+
+      int main(int argc, char **argv) {
+        if (argc < 2) {
+          printf("Tokenize filename [options ...]\\n");
+          exit(1);
+        }
+
+        const auto filename = argv[1];
+        const auto cmdArgs = &argv[2];
+        auto numArgs = argc - 2;
+
+        CXIndex index = clang_createIndex(1, 0);
+        CXTranslationUnit tu = clang_parseTranslationUnit(index, filename, cmdArgs, numArgs, NULL, 0, 0);
+        if (tu == NULL) {
+          printf("Cannot parse translation unit\\n");
+          return 1;
+        }
+
+        CXSourceRange range = get_filerange(tu, filename);
+        CXToken *tokens;
+        unsigned numTokens;
+        clang_tokenize(tu, range, &tokens, &numTokens);
+        printf("NumTokens: %d\\n", numTokens);
+
+        clang_disposeTokens(tu, tokens, numTokens);
+        clang_disposeTranslationUnit(tu);
+        clang_disposeIndex(index);
+        return 0;
+      }
+    EOS
```

Comment:
Can we maybe trim down that example C source a bit?

Reply:
Done.